### PR TITLE
Coloring based on file extension

### PR DESCRIPTION
**Pull request by jozsefsallai**
The color of the extension and the left border of the list items will be different based on the extension of the file. Unit test included.

### CODE REVIEW

File: ui/src/components/dashboard/audio_list_item.test.coffee [hunk 1]
```diff
@@ -54,6 +54,25 @@ describe 'audio-list-item', ->
   it 'should not have an invisible icon', ->
     expect(@wrapper.find('.fa.fa-eye-slash')).to.have.length 0
 
+  describe 'list item accent color', ->
+    extension_check = (ext, color) =>
+      it 'should be '+color+' on '+ext, ->
```
Comment (austindebruyn) [inline ui/src/components/dashboard/audio_list_item.test.coffee]:
```coffee
it "should be #{color} on #{ext}" ->
```

File: ui/src/components/dashboard/audio_list_item.test.coffee [hunk 2]
```diff
@@ -54,6 +54,25 @@ describe 'audio-list-item', ->
...
+        new_fixture = Object.assign {}, audios_fixture.chicken, url: 'chicken.' + ext
```
Comment (austindebruyn) [inline ui/src/components/dashboard/audio_list_item.test.coffee]: same string interpolation - `"chicken.#{ext}"`

File: ui/src/components/dashboard/audio_list_item.test.coffee [hunk 3]
```diff
@@ -54,6 +54,25 @@ describe 'audio-list-item', ->
...
+        propsData = 
+          audio: new_fixture
+        @wrapper = mount audio_list_item, globals: ($store: @store), propsData:
+          audio: new_fixture
```
Comment (austindebruyn) [inline ui/src/components/dashboard/audio_list_item.test.coffee]: duplicate... propsData on the line above is unused

File: ui/src/components/dashboard/audio_list_item.test.coffee [hunk 4]
```diff
@@ -54,6 +54,25 @@ describe 'audio-list-item', ->
...
+        expect(@wrapper.first('.dashboard-audio-list-item').hasClass(ext)).to.be.true
+
+    extension_check 'mp3', 'blue'
+    extension_check 'wav', 'red'
+    extension_check 'flac', 'pink'
+    extension_check 'aif', 'green'
+    extension_check 'wma', 'yellow'
+    extension_check 'mid', 'purple'
+    extension_check 'midi', 'purple'
+    extension_check 'unknown', 'gray'
```
Comment (austindebruyn) [inline ui/src/components/dashboard/audio_list_item.test.coffee]: no real need for the second argument at all, yeah? it's not used by the test and is one more thing to change if you want to switch up with the colors

File: ui/src/components/dashboard/audio_list_item.test.coffee (hunk 1 above)
Comment (jozsefsallai) [reply ui/src/components/dashboard/audio_list_item.test.coffee]: Yeah, good idea, I just personally like to stick to one either single quotes or double quotes :) Will do this though!

File: ui/src/components/dashboard/audio_list_item.test.coffee (hunk 3 above)
Comment (jozsefsallai) [reply ui/src/components/dashboard/audio_list_item.test.coffee]: Whoops, my bad! This was from one of my failed attempts, forgot to delete 🤦‍♂️

File: ui/src/components/dashboard/audio_list_item.test.coffee (hunk 4 above)
Comment (jozsefsallai) [reply ui/src/components/dashboard/audio_list_item.test.coffee]: Hmm, fair enough. It was used in the `it`s, but yeah, I also think it's pretty unnecessary.

**Comment by jozsefsallai** [comment]
Also made the unit test look nicer by putting the file formats into an array and iterating through it

**Comment by austindebruyn** [comment]
little skeptical of the linter rule overrides. is there a pressing reason? also there's a concern that you're passing dirty user input right to css. I could name my file `anything.d-table` and get a `<div class="dashboard-audio-list-item d-table">` which blows up the page layout. any ideas?

**Comment by jozsefsallai** [comment]
Unsure why Coffeelint thinks the fat arrow is unnecessary in this situation, it clearly doesn't work otherwise. I added the override to avoid this issue in the UI lint.

As for the other point, fair enough, I was actually worried about that myself. Maybe if we change the classes to something like `extension-mp3`?

**Comment by austindebruyn** [comment]
I'm down with the prefix. can you try `# coffeelint: disable=whatever_rule` to hush coffeelint? I prefer inline overrides than globals

**Comment by jozsefsallai** [comment]
Yep!

**Pull request merged**